### PR TITLE
Enable Python3 checker messages for dict.iter*()

### PR DESCRIPTION
**Pull request by tiagohonorato**
Dict methods iteritems(), iterkeys() and itervalues() are longer supported
in Python 3. The check was disabled for some lines as AvocadoParam,
AvocadoParams and ValueDict classes define a method called iteritems()
that triggers the warning.

Related to #3466

Signed-off-by: Tiago Honorato <tiagohonorato1@gmail.com>

### CODE REVIEW

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @tiagohonorato LGTM, just can you please resolve the conflicts which formed after the merge of your last PR #4405

**Comment by tiagohonorato** [comment]
Hi @richtja, thanks for pointing that out. I will resolve the conflicts.

**Review by richtja — APPROVED** [review]
LGTM, thanks

**Review by willianrampazzo — COMMENTED** [review]
This PR fixes 3 places and adds local disables in 8 places. I'm wondering if it is worth enabling the check globally without fixing most of the places we have problems with now.

**Comment by tiagohonorato** [comment]
Hi @willianrampazzo, the check was disabled in some places because I see those as false positives. To issue the warning, Pylint seems to just check whether methods named ``iterkeys()``, ``itervalues()`` and ``iteritems()`` are called, not taking the instance in consideration, in other words, if it is a ``dict`` (I will take a closer look at Pylint's codebase and open an issue there if it is confirmed). Consequently, if we define a method named ``iterkeys()``, ``itervalues()`` or ``iteritems()`` it will trigger the error when called, even though we are not referring to an instance of ``dict``, but to, for example, an instance of ``AvocadoParam``.

On the other hand, I see your point and it might be useful to change these names that conflict with ``dict.iter*``methods to avoid confusion when the check is enabled, specially for new code that uses ``AvocadoParam`` and ``AvocadoParams``. Maybe change these methods to something like ``iter_items()``, what do you think?

File: optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/mux.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def iteritems(self):
 
     def items(self):
         """ Slower implementation with the use of __getitem__ """
-        return self.iteritems()
+        return self.iteritems()  # pylint: disable=W1620
```
Comment (tiagohonorato) [inline optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/mux.py:297]: By the way, after reviewing the modifications after @willianrampazzo's comment, we might me able to remove this method entirety, as ``ValueDict`` inherits from ``dict`` and the return from ``dict.items`` is already an iterator.

**Comment by richtja** [comment]
> Hi @willianrampazzo, the check was disabled in some places because I see those as false positives. To issue the warning, Pylint seems to just check whether methods named `iterkeys()`, `itervalues()` and `iteritems()` are called, not taking the instance in consideration, in other words, if it is a `dict` (I will take a closer look at Pylint's codebase and open an issue there if it is confirmed). Consequently, if we define a method named `iterkeys()`, `itervalues()` or `iteritems()` it will trigger the error when called, even though we are not referring to an instance of `dict`, but to, for example, an instance of `AvocadoParam`.
> 
> On the other hand, I see your point and it might be useful to change these names that conflict with `dict.iter*`methods to avoid confusion when the check is enabled, specially for new code that uses `AvocadoParam` and `AvocadoParams`. Maybe change these methods to something like `iter_items()`, what do you think?

@willianrampazzo, @tiagohonorato about this problem I already opened discussion #4414, where we can find out the best approach.

**Comment by willianrampazzo** [comment]
> > Hi @willianrampazzo, the check was disabled in some places because I see those as false positives. To issue the warning, Pylint seems to just check whether methods named `iterkeys()`, `itervalues()` and `iteritems()` are called, not taking the instance in consideration, in other words, if it is a `dict` (I will take a closer look at Pylint's codebase and open an issue there if it is confirmed). Consequently, if we define a method named `iterkeys()`, `itervalues()` or `iteritems()` it will trigger the error when called, even though we are not referring to an instance of `dict`, but to, for example, an instance of `AvocadoParam`.
> > On the other hand, I see your point and it might be useful to change these names that conflict with `dict.iter*`methods to avoid confusion when the check is enabled, specially for new code that uses `AvocadoParam` and `AvocadoParams`. Maybe change these methods to something like `iter_items()`, what do you think?
> 
> @willianrampazzo, @tiagohonorato about this problem I already opened discussion #4414, where we can decide the best approach.

Thanks, @richtja!

**Comment by willianrampazzo** [comment]
> Hi @willianrampazzo, the check was disabled in some places because I see those as false positives. To issue the warning, Pylint seems to just check whether methods named `iterkeys()`, `itervalues()` and `iteritems()` are called, not taking the instance in consideration, in other words, if it is a `dict` (I will take a closer look at Pylint's codebase and open an issue there if it is confirmed). Consequently, if we define a method named `iterkeys()`, `itervalues()` or `iteritems()` it will trigger the error when called, even though we are not referring to an instance of `dict`, but to, for example, an instance of `AvocadoParam`.
> 
> On the other hand, I see your point and it might be useful to change these names that conflict with `dict.iter*`methods to avoid confusion when the check is enabled, specially for new code that uses `AvocadoParam` and `AvocadoParams`. Maybe change these methods to something like `iter_items()`, what do you think?

I see your point and I agree it makes sense.

**Review by willianrampazzo — APPROVED** [review]

**Review by beraldoleal — CHANGES_REQUESTED** [review]
@tiagohonorato, thanks for this. LGTM, but my only suggestion is to rebase this on top of the master and remove the "merge" commit.

**Comment by tiagohonorato** [comment]
@beraldoleal, sure!

**Comment by tiagohonorato** [comment]
I decided to keep ``ValueDict.iteritems()`` and ``ValueDict.items()`` here, as that may need further discussion and keeping those shouldn't cause any problems.

**Review by beraldoleal — APPROVED** [review]

**Pull request merged**